x-pack/plugins/security_solution/public/common/components/events_viewer/selectors/index.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createSelector } from 'reselect';
+
+import {
+  getTimelineSelector,
+  globalFiltersQuerySelector,
+  globalQuery,
+  globalQuerySelector,
+  timelineQueryByIdSelector,
+} from '../../../store/inputs/selectors';
+import { getTimelineByIdSelector } from '../../../../timelines/store/timeline/selectors';
+
+/**
+ * This selector is invoked with two arguments:
+ * @param state - the state of the store as defined by State in common/store/types.ts
+ * @param id - a timeline id e.g. `detections-page`
+ *
+ * Example:
+ *  `useSelector((state: State) => eventsViewerSelector(state, id))`
+ */
+export const eventsViewerSelector = createSelector(
```

Comment:
This combines all of the individual selectors previously used by the `StatefulEventsViewer` into a single selector via [reselect](https://github.com/reduxjs/reselect)'s [createSelector](https://github.com/reduxjs/reselect#createselectorinputselectors--inputselectors-resultfunc-selectoroptions) API